etcdserver/v3_server.go

```diff
@@ -464,6 +466,15 @@ func (s *EtcdServer) Authenticate(ctx context.Context, r *pb.AuthenticateRequest
 }
 
 func (s *EtcdServer) UserAdd(ctx context.Context, r *pb.AuthUserAddRequest) (*pb.AuthUserAddResponse, error) {
+	if r.Options == nil || !r.Options.NoPassword {
+		hashedPassword, err := bcrypt.GenerateFromPassword([]byte(r.Password), s.authStore.BcryptCost())
+		if err != nil {
+			return nil, err
+		}
+		r.HashedPassword = base64.StdEncoding.EncodeToString(hashedPassword)
+		r.Password = ""
```

Comment:
if old version member(< 3.5) receive a UserAdd request from new version member(r.Password is empty), it will also cause data inconsistency.
it seems we have to set r.Password to empty when all member version is 3.5+.

Reply:
Good point, but this PR changes the behavior of the state machine so the cluster older than 3.4 cannot be rolling upgraded 3.5 (note that shutting down the existing cluster and upgrading all of the members at once is ok). We don't have a good way to do rolling upgrade (or downgrade) etcd when this kind of change happens.

Reply:
I'll add this point to the doc.

Reply:
After the user upgrades cluster from 3.4 to 3.5, it is not safe to downgrade the cluster version from 3.5 to 3.4 due to this feature.  /cc @YoyinZyc

Reply:
> I'll add this point to the doc.

thanks.

Reply:
> After the user upgrades cluster from 3.4 to 3.5, it is not safe to downgrade the cluster version from 3.5 to 3.4 due to this feature. /cc @YoyinZyc

Downgrade feature also changes the backend api which means we cannot downgrade from 3.5 to 3.4 and the first downgradable version is the next minor version of 3.5.  Since this feature is added in 3.5, then it will not affect downgrade.

Reply:
Added the description to the docs in the 2nd commit, could you take a look @tangcong @YoyinZyc ?

Reply:
> Added the description to the docs in the 2nd commit, could you take a look @tangcong @YoyinZyc ?

LGTM. Thanks!

Reply:
Thanks @YoyinZyc !